lib/server/monalisa-synchronization/MonALISAInterface.js

```diff
@@ -0,0 +1,169 @@
+     * @return {DataPass[]} data passes
+     */
+    async getDataPasses() {
+        const payload = await this.fetchDataPasses();
+        return this.processDataPassesPayload(payload);
+    }
+
+    /**
+     * Fetch data passes from MonALISA
+     * @return {object<string, Partial<MonalisaDataPass>>} payload
+     */
+    async fetchDataPasses() {
+        const { default: fetch } = await import('node-fetch');
+        const specificURL = new URL(this.dataPassesUrl);
+        specificURL.searchParams.set('res_path', 'json');
+        return fetch(specificURL, this.httpOptions).then((response) => response.json());
+    }
+
+    /**
+     * Adapt data pass
+     * @param {MonALISADataPass} dataPass data pass
+     * @return {DataPass} adapted data pass
+     */
+    monalisaDataPassToEntity(dataPass) {
+        return {
+            name: dataPass.name,
+            outputSize: dataPass.output_size ? Number(dataPass.output_size) : null,
+            reconstructedEventsCount: dataPass.reconstructed_events,
+            description: dataPass.description,
+            lastRunNumber: dataPass.last_run,
+        };
+    }
+
+    /**
+     * Process server response payload to database entites, remove unnecessary data
+     * @param {object<string, Partial<MonalisaDataPass>>} payload raw data acquired from MonALISA: mapping of Data Pass Name to its parameters
+     * @return {DataPass[]} adapted data
+     */
+    processDataPassesPayload(payload) {
+        return Object.entries(payload)
+            .map(([dataPassName, dataPassParamenters]) => ({
+                name: dataPassName.trim(),
+                ...dataPassParamenters,
+            }))
+            .filter((dataPass) => extractLhcPeriod(dataPass.name)?.year >= this.yearLowerLimit)
+            .map((dataPass) => this.monalisaDataPassToEntity(dataPass));
+    }
+
+    /**
+     * Get data pass details
+     * @param {string} description description property of a data pass
+     * @return {DataPassDetails} data pass details
+     */
+    async getDataPassDetails(description) {
+        const payload = await this.fetchDataPassDetails(description);
+
+        return this.processDataPassDetailsPayload(payload);
+    }
+
+    /**
+     * Fetch data pass properties from MonALISA
+     * @param {string} description description property of a data pass
+     * @return {object<number, { run_no: number }>} payload
+     */
+    async fetchDataPassDetails(description) {
+        const { default: fetch } = await import('node-fetch');
+        const specificURL = new URL(this.dataPassDetailsUrl);
+        specificURL.searchParams.set('res_path', 'json');
+        specificURL.searchParams.append('filter_jobtype', description);
+        return fetch(specificURL, this.httpOptions).then((response) => response.json());
+    }
+
+    /**
+     * Process server response payload to database entites, remove unnecessary data
+     * @param {object<number, { run_no: number }>} payload payload
```

Comment:
What is payload? => MonALISA data pass details endpoint response payload

Reply:
applied